docs/features.md

```diff
@@ -59,53 +59,35 @@ References:
 #### Since: generic-worker 10.6.0
 
 The taskcluster proxy provides an easy and safe way to make authenticated
-taskcluster requests within the scope(s) of a particular task.
+taskcluster requests within the scope(s) of a particular task.  The proxy
+accepts un-authenticated requests and attaches credentials to them
+corresponding to `task.scopes` as well as scopes to upload artifacts.
 
-For example lets say we have a task like this:
+The proxy's rootUrl is available to tasks in the environment variable
+`TASKCLUSTER_PROXY_URL`.  It can be used with a client like this:
 
 ```js
-{
-  "scopes": ["a", "b"],
-  "payload": {
-    "features": {
-      "taskclusterProxy": true
-    }
-  }
-}
+var taskcluster = require('taskcluster-client');
+var queue = new taskcluster.Queue({
+  rootUrl: process.env.TASKCLUSTER_PROXY_URL,
+});
+queue.createTask(..);
 ```
 
-A web service will execute (typically on port 80) of the local machine for the
-duration of the task, with which you can proxy unauthenticated requests to
-various taskcluster services. The proxy will inject the Authorization http
-header for you and proxy the request to the target service, granting the
-request the scopes of the task (in this case ["a", "b"]).
+This request would require that `task.scopes` contain the appropriate
+`queue:create-task:..` scope for the `createTask` API call.
 
-| Target Destination                             | Proxy Address                            |
-|------------------------------------------------|------------------------------------------|
-| https://queue.taskcluster.net/<PATH>           | http://localhost/queue/<PATH>            |
-| https://index.taskcluster.net/<PATH>           | http://localhost/index/<PATH>            |
-| https://aws-provisioner.taskcluster.net/<PATH> | http://localhost/aws-provisioner/<PATH>  |
-| https://secrets.taskcluster.net/<PATH>         | http://localhost/secrets/<PATH>          |
-| https://auth.taskcluster.net/<PATH>            | http://localhost/auth/<PATH>             |
-| https://hooks.taskcluster.net/<PATH>           | http://localhost/hooks/<PATH>            |
-| https://purge-cache.taskcluster.net/<PATH>     | http://localhost/purge-cache/<PATH>      |
+*NOTE*: as a special case, the scopes required to call
+`queue.createArtifact(<taskId>, <runId>, ..)` are automatically included,
+regardless of `task.scopes`.
 
-For example (using curl) inside a task container.
+The proxy is easy to use within a shell command, too:
 
 ```sh
-cat secret | curl --header 'Content-Type: application/json' --request PUT --data @- http://localhost/secrets/v1/secret/<secretName>
+curl $TASKCLUSTER_PROXY_URL/api/secrets/v1/secret/my-top-secret-secret
```

Comment:
I prefer the example with PUT as it demonstrates that the proxy is not only for fetching data, but for all taskcluster endpoints.